server/sonar-web/src/main/webapp/WEB-INF/app/controllers/project_controller.rb

```diff
@@ -232,7 +232,7 @@ def delete_snapshot_history
 
     sid = params[:snapshot_id]
     if sid
-      Snapshot.update_all("status='U'", ["id=? or root_snapshot_id=(?)", sid.to_i, sid.to_i])
+      Snapshot.update_all("status='U'", ["id=?", sid.to_i])
```

Comment:
yes, a single `update` can be used. 
Using uuid requires to change view too. I'm not confident in refactoring that. That should be handled during the rewriting in Java.